### ISSUE TITLE
Issue with Edge on Win10.

### ISSUE THREAD

**Issue by spinor8** [mliebelt/PgnViewerJS#115]
PgnViewerJS seems to have an issue with Edge on Windows10. Here's how to reproduce it.

On a Windows 10 machine, download the PgnViewerJS directory. Open examples/view.html in Edge. I see a completely blank page.

To try to debug, click on F12 to go to Developer Tools. I am seeing this error
`SCRIPT16389: SCRIPT16389: Unspecified error.     pgnvjs.js (1947,7)`

When I click on the error location. I see
`if (!window.localStorage || !lngs.length) {`

Nothing our of the ordinary. I tried opening the same file with Firefox on the same machine. It works without any issues. What's responsible for this? Thanks for any insight.

**Comment by mliebelt** [comment]
Thank you for the feedback. I don't car too much about IE11 any more, but Edge should work. Due to my current work computer (Linux based), I will try to find the time to check that on a VirtualBox image and see if I understand the problem. There should be at least a polyfill available.

**Comment by gbtami** [comment]
http://mliebelt.github.io/PgnViewerJS/docu/examples.html seems OK for me on Win10 and Microsoft Edge 41.16299.967.0

**Comment by spinor8** [comment]
@gbtami The website is fine but not the actual repo. I downloaded the zipped file, then unzipped everything. On the same Edge browser running on Win10 machine, I am seeing the local examples/view.html showing blank (as I reported earlier) but the website showing fine.

**Comment by mliebelt** [comment]
But that is ok. So by downloading the repo, you just get some development state (which may be ok or not). I will check if I find the error, but other things are currently more urgent ...

**Comment by spinor8** [comment]
@mliebelt Thanks, take your time. I was experiencing the issue with Edge from the 0.9.7 release zip file I downloaded, https://s3.eu-central-1.amazonaws.com/pgnviewerjs/releases/PgnViewerJS-0.9.7.zip.

**Comment by mliebelt** [comment]
So I tried to reproduce it, and found that `view.html` is part of the distribution. This file is there all the time, but the file `pgnvjs.js` is generated by the build process, before I do a release. I tried it with all installed browsers i could find (Edge, IE11, Chrome on Windows, Brave, Chromium, Firefox on Linux), and it works for all with the exception of IE11 (which I don't care ...).

When you read the README carefully, there is a section that is named "Deployment > From Clone". There you will find:

> Clone the repository.
> Run npm install from the root, to ensure are tools are installed.
> Run grunt default to build a distribution.

This ensures that a current `pgnvjs.js` file is generated, that fits then the using file `dist/examples/view.html`.

One additional caveat: This software is used on a web server, so you may face not working parts when you just open an HTML file without running a web server. When I check my examples, I do the following:

* Run a local web server in that directory (see for example [NodeJS - run a local web server](https://jasonwatmore.com/post/2016/06/22/nodejs-setup-simple-http-server-local-web-server) how to do that (you need node.js anyway when developing)).
* Open the example directory, and navigate to the file.
* Open the file in your web browser.

Please give me some feedback, if that solves the problem for you. Working with any current state needs a local build before you can use any examples.

**Comment by spinor8** [comment]
@mliebelt Thanks for taking a look at it. I was using the viewer for local files not via the web server. I guess this is an unsupported use case.

**Comment by mliebelt** [comment]
@spinor8 Checked your case again, and perhaps the problem exists really in the distribution. I checked the following:

* Downloading the whole repository, from the tag 0.9.7 ==> When building the distribution version (`npm install && grunt default`), the directory `examples` is missing from the distribution.
* Downloading the zip file from the github website (https://s3.eu-central-1.amazonaws.com/pgnviewerjs/releases/PgnViewerJS-0.9.7.zip), unzipping it locally ==> Seems to work for most browsers, some tell me they have CORS problems. When using a locally started web server, everything is ok in all browsers (with the exception of the `favicon.ico`).

So perhaps I was a little bit sloppy when building the distribution, and corrected the downloadable file after having made the tag (and never changed it). Please go and use the new version 0.9.8, I have checked that version now numerous times, to ensure that it works out of the box.

**Comment by spinor8** [comment]
@mliebelt Thanks for taking another look at the issue. I have downloaded the version 0.9.8 and running it just locally (without a web server), I am afraid that I am seeing a similar issue on Edge (Chrome and Firefox both work) in Windows10 when I open the view.html. The line number has shifted.
```
SCRIPT16389: SCRIPT16389: Unspecified error.                                        pgnvjs.js (4057,7)
```
When clicked on that particular line in the error, it shows once again
```
      if (!window.localStorage || !lngs.length) {
```
This is the same code snippet that Edge didn't like in 0.9.7

I think Edge really doesn't like the window.localStorage. 

I googled "windows localstorage edge" and came across this issue raised on Microsoft's developer website.
https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/8816771/

Guess it won't run on Edge for local files.
